protocol/pubsub/v2/internal/connection.go

```diff
@@ -89,8 +93,35 @@ const (
 var DefaultReceiveSettings = pubsub.ReceiveSettings{
 	// Pubsub default receive settings will fill in other values.
 	// https://godoc.org/cloud.google.com/go/pubsub#Client.Subscription
+}
 
-	Synchronous: false,
+// pubsub/v2 dropped Exists methods and suggests optimistically using GetTopic instead
+func topicExists(ctx context.Context, client *pubsub.Client, topicID string) (bool, error) {
+	// Check if the topic exists.
+	_, err := client.TopicAdminClient.GetTopic(ctx, &pubsubpb.GetTopicRequest{
+		Topic: fmt.Sprintf("projects/%s/topics/%s", client.Project(), topicID),
+	})
+	if err != nil {
+		if st, ok := status.FromError(err); ok && st.Code() == codes.NotFound {
+			return false, nil // Topic does not exist.
+		}
+		return false, fmt.Errorf("unable to get topic %q, %v", topicID, err)
```

Comment:
nit: use `%w` for error wrapping?

Reply:
good idea but since the calling function bubbles-up any caught errors and wraps them again with %v, it would need changes there too. So I'll leave it as is as its likely not used with `errors.Is` anywhere or would have caused issues before